drivers/sensor/tmp116/Kconfig

```diff
@@ -8,3 +8,10 @@ config TMP116
 	depends on I2C
 	help
 	  Enable driver for TMP116 and TMP117 temperature sensors.
+
+	config SENSOR_TMP116_INIT_PRIORITY
```

Comment:
Is this really needed? I've been working on reducing the number of driver-specific init priority configs and instead using common configs for each driver class (e.g., SENSOR_INIT_PRIORITY, EEPROM_INIT_PRIORITY, etc.)

Reply:
out of curiosity what is the reason for limiting those driver-specific priorities ? 
Usually for my applications I end up with a lot of driver-specific init priority defined otherwise because of devices dependency some initialization fails. That said it's probably less true for "leaf" devices like sensors or eeprom. I will review the usage.

In any case I will change the default value of `EEPROM_TMP116_INIT_PRIORITY` to `EEPROM_INIT_PRIORITY`

Reply:
I think this config is not really required in my application so I will remove it as per your suggestion. `EEPROM_INIT_PRIORITY` doesn't seem to exist so it's propably why I set `EEPROM_TMP116_INIT_PRIORITY` default to an arbitrary value

Reply:
Please see #40044 which adds `EEPROM_INIT_PRIO`.

This cleanup of initialization priorities was triggered by #38631. Many drivers use `KERNEL_INIT_PRIORITY_DEVICE` directly, which can cause problems if you need to initialize two drivers in a particular order, in that case it was a clock driver and a UART driver. We needed a little more granularity to specify initialization priorities, but didn't want to introduce the Kconfig symbol sprawl that would come with specifying priorities for every single driver. It will also make applications more portable across multiple boards.

Reply:
@MaureenHelm I rebased on main after  #40044 was merged and removed the driver specific init priority in favor of `EEPROM_INIT_PRIORITY`